ckanext/canada/cli.py

```diff
@@ -1900,3 +1163,64 @@ def _drop_function(name, verbose=False):
             click.echo('Failed to drop function: {0}\n{1}'.format(
                 name, str(datastore._programming_error_summary(pe))), err=True)
         pass
+
+
+def _maybe_init_harvest_db(verbose=True):
+    # check to see if the harvet_object table exists.
+    # bad workaround, but subclassing a plugin does not allow
+    # us to run migrations on the parent plugin.
+    try:
+        model.Session.query(HarvestObject).first()
+        if verbose:
+            _success_message('Harvest database tables already exist.')
+    except ProgrammingError as e:
+        if verbose:
+            _error_message('Harvest database tables not initialized, creating them now...')
+        if 'psycopg2.errors.UndefinedTable' in str(e):
+            if plugin_loaded('canada_harvest'):
+                unload('canada_harvest')
+            if not plugin_loaded('harvest'):
+                load('harvest')
+
+            _run_migrations(plugin='harvest', version='head')
+
+            if plugin_loaded('harvest'):
+                unload('harvest')
+            if not plugin_loaded('canada_harvest'):
+                load('canada_harvest')
+            if verbose:
+                _success_message('Harvest database tables initialized.')
+        pass
+
+
+@canada.command(short_help="Creates harvest database tables.")
+@click.option('-f', '--refresh', is_flag=True, type=click.BOOL, help='Forces the refresh of all the source objects in the database.')
+@click.option('-q', '--quiet', is_flag=True, type=click.BOOL, help='Suppresses human inspection.')
+def init_harvest_plugin(refresh=False, quiet=False):
+    """Creates harvest database tables."""
+
+    _maybe_init_harvest_db()
```

Comment:
can we not https://docs.ckan.org/en/2.10/extensions/best-practices.html#use-migrations-when-introducing-new-models or is that just not how the harvest extension works?

Reply:
The parent Harvest plugin does use that. The issue might be a CKAN Core thing, or at least a dependency thing (whichever dependency is used for the migration stuff).

So if you sub-class a plugin, and try to run the migrations, e.g. in this case `ckan db upgrade -p canada_harvest` it is not going to work. It raises a bunch of errors having to do with the name of the plugin and the directory of the migration scripts. But doing `ckan db upgrade -p harvest` does work. Which is what this crazy CLI command does programatically, it unloads `canada_harvest`, loads the parent `harvest` plugin and then runs the migrations for `harvest`.